packages/state-transition/src/block/processWithdrawals.ts

```diff
@@ -1,25 +1,73 @@
 import {ssz, capella} from "@lodestar/types";
-import {MAX_WITHDRAWALS_PER_PAYLOAD} from "@lodestar/params";
+import {
+  MAX_EFFECTIVE_BALANCE,
+  WITHDRAWAL_PREFIX_BYTES,
+  ETH1_ADDRESS_WITHDRAWAL_PREFIX,
+  MAX_WITHDRAWALS_PER_PAYLOAD,
+} from "@lodestar/params";
+import {byteArrayEquals} from "@chainsafe/ssz";
 
 import {CachedBeaconStateCapella} from "../types.js";
+import {decreaseBalance} from "../util/index.js";
+
+export function getExpectedWithdrawals(state: CachedBeaconStateCapella): capella.Withdrawal[] {
+  const currentEpoch = state.epochCtx.epoch + 1;
+  let withdrawalIndex = state.nextWithdrawalIndex;
+  let validatorIndex = state.latestWithdrawalValidatorIndex;
+  const {validators, balances} = state;
+
+  const withdrawals: capella.Withdrawal[] = [];
+  for (let index = 0; index < validators.length; index++) {
+    // Get the index of validator next in turn
+    validatorIndex = (validatorIndex + 1) % validators.length;
+    const validator = validators.get(validatorIndex);
+    const balance = balances.get(validatorIndex);
+    const {effectiveBalance, withdrawalCredentials, withdrawableEpoch} = validator;
+
+    if (
+      ((balance > 0 && withdrawableEpoch <= currentEpoch) ||
+        (effectiveBalance === MAX_EFFECTIVE_BALANCE && balance > MAX_EFFECTIVE_BALANCE)) &&
+      byteArrayEquals(withdrawalCredentials.slice(0, WITHDRAWAL_PREFIX_BYTES), ETH1_ADDRESS_WITHDRAWAL_PREFIX)
```

Comment:
Since `WITHDRAWAL_PREFIX_BYTES` is 1,
worth turning this into something like
`withdrawalCredentials[0] === ETH1_SOMETHING`

Reply:
makes sense :+1: